<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Fig 4</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">delfipro2024</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Fig 4</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>delfipro2024/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsdelfipro2024tree0bc9e1d3a5845eafd155da362449ca99bfdbad88targetblank0bc9e1da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/cancer-genomics/delfipro2024/tree/0bc9e1d3a5845eafd155da362449ca99bfdbad88" target="_blank">0bc9e1d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsdelfipro2024tree0bc9e1d3a5845eafd155da362449ca99bfdbad88targetblank0bc9e1da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/cancer-genomics/delfipro2024/tree/0bc9e1d3a5845eafd155da362449ca99bfdbad88" target="_blank">0bc9e1d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  README.html
    Untracked:  renv.lock
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/S9.Rmd</code>) and HTML
(<code>docs/S9.html</code>) files. If youâ€™ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/delfipro2024/28c6a1bb309a862e2e60fbf2bdc779820bea10a0/docs/S9.html" target="_blank">28c6a1b</a>
</td>
<td>
shay-279
</td>
<td>
2024-08-01
</td>
<td>
aa fix more stuff
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/delfipro2024/blob/c74d1ee3872c077b7bd65f0ffb6aa657b10a8e94/analysis/S9.Rmd" target="_blank">c74d1ee</a>
</td>
<td>
shay-279
</td>
<td>
2024-08-01
</td>
<td>
start
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/delfipro2024/c74d1ee3872c077b7bd65f0ffb6aa657b10a8e94/docs/S9.html" target="_blank">c74d1ee</a>
</td>
<td>
shay-279
</td>
<td>
2024-08-01
</td>
<td>
start
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Read in Noushinâ€™s data for TCGA figure</p>
<pre class="r"><code>fig.data &lt;- readRDS(here(&#39;data&#39;, &#39;tcga&#39;, &#39;data.rds&#39;))</code></pre>
<div id="tcga-copy-number" class="section level2">
<h2>TCGA copy number</h2>
<p>TCGA figure panel â€“ problems, chromosome widths, making the panel
match size of one facet in z-score panel</p>
<pre class="r"><code>setDT(fig.data)
fig.data[,bin:=as.factor(rev(bin))][]</code></pre>
<pre><code>     seqnames      pos    arm    bin   variable        value disease change
       &lt;fctr&gt;    &lt;num&gt; &lt;fctr&gt; &lt;fctr&gt;     &lt;fctr&gt;        &lt;num&gt;  &lt;char&gt; &lt;char&gt;
  1:     chr1  7500000     1p    473 liver.Loss -0.208754209   liver   Loss
  2:     chr1 17500000     1p    472 liver.Loss -0.239057239   liver   Loss
  3:     chr1 22500000     1p    471 liver.Loss -0.254208754   liver   Loss
  4:     chr1 27500000     1p    470 liver.Loss -0.227272727   liver   Loss
  5:     chr1 32500000     1p    469 liver.Loss -0.126262626   liver   Loss
 ---                                                                       
942:    chr22 27500000    22q      5 liver.Gain  0.028619529   liver   Gain
943:    chr22 32500000    22q      4 liver.Gain  0.037037037   liver   Gain
944:    chr22 37500000    22q      3 liver.Gain  0.020202020   liver   Gain
945:    chr22 42500000    22q      2 liver.Gain  0.006734007   liver   Gain
946:    chr22 47500000    22q      1 liver.Gain  0.000000000   liver   Gain</code></pre>
<pre class="r"><code>fig.data$disease&lt;-&quot;TCGA Ovarian Cancers\n(n = 597)&quot;
plot &lt;- fig.data %&gt;%
    mutate(disease=&quot;TCGA Ovarian Cancers\n(n = 597)&quot;) %&gt;%
    ggplot(aes(x=value, y=bin, color=change, fill=change)) +
    facet_grid(arm~disease, scales = &#39;free_y&#39;) + ##,
    ##           switch = &#39;y&#39;)+#, space = &#39;free&#39;)+
    ##facet_wrap(~arm, ncol=1, scales=&quot;free_y&quot;) + ##
    geom_col() +
    scale_x_continuous() +
    #scale_fill_brewer(palette = &#39;Set1&#39;)+
    #scale_color_brewer(palette = &#39;Set1&#39;)+
    scale_color_manual(values=c(&quot;red3&quot;,&quot;#18468B&quot;))+
    scale_fill_manual(values=c(&quot;red3&quot;,&quot;#18468B&quot;))+
    theme_minimal(base_size=24) +
    theme(strip.text.y = element_text(angle = 0,
                                      hjust = 0.5,
                                      vjust = 0.5,
                                      size=20),
          strip.text.x=element_text(size=20),
          axis.text.y = element_blank(),
          axis.ticks = element_blank(),
          panel.grid =  element_blank(),
          legend.title = element_blank(),
          legend.position=&quot;bottom&quot;,
          axis.title.x = element_blank(),
          strip.background = element_blank()) + ##removing label
    #xlab(&quot;Proportion of\ncases with CNV&quot;) +
    ylab(&quot;Chromosome Arm&quot;)</code></pre>
</div>
<div id="ovarian-study-z-scores" class="section level2">
<h2>Ovarian study z-scores</h2>
<p>The z-scoreâ€™s with our data panel. Problems â€“ chromosome width.</p>
<pre class="r"><code>meta&lt;-fread(here(&quot;data&quot;,&quot;metadata.csv&quot;))
meta&lt;-meta %&gt;% select(-V1) %&gt;% mutate(classifier_type2=if_else(id==&quot;CGPLOV161P&quot;,&quot;Benign&quot;,classifier_type2))

data&lt;-fread(here(&quot;data&quot;,&quot;screening_train_data.csv&quot;)) %&gt;% select(-type)
data2&lt;-fread(here(&quot;data&quot;,&quot;diag_train_data.csv&quot;)) %&gt;% filter(type==&quot;healthy&quot;) %&gt;% select(-type)
data&lt;-rbind(data,data2)

data&lt;-inner_join(data,meta %&gt;% select(id,classifier_type),by=&quot;id&quot;) %&gt;% rename(type=classifier_type)



features&lt;-data

cols &lt;- brewer.pal(7,&quot;Set1&quot;)
cols[3] &lt;- cols[4]

features &lt;- features %&gt;% select(id,starts_with(&quot;z&quot;))
features &lt;- inner_join(features, meta %&gt;%
                                     select(id,`classifier_type2`),
                           by=c(&quot;id&quot;=&quot;id&quot;))




datx&lt;-read_csv(here(&quot;data/scores&quot;,&quot;All_Scores_CV.csv&quot;))</code></pre>
<pre><code>New names:
Rows: 479 Columns: 22
â”€â”€ Column specification
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ Delimiter: &quot;,&quot; chr
(16): id, id2, cohort, type, type2, training_set_include_screening, trai... dbl
(5): ...1, Screening ZEUS, Diagnostic ZEUS, ca125, he4 lgl (1):
menopause_status
â„¹ Use `spec()` to retrieve the full column specification for this data. â„¹
Specify the column types or set `show_col_types = FALSE` to quiet this message.
â€¢ `` -&gt; `...1`</code></pre>
<pre class="r"><code>datx&lt;-datx %&gt;% filter(training_set_include_screening==&quot;yes&quot;|training_set_include_diag==&quot;yes&quot;)
datx&lt;-datx %&gt;% select(id,classifier_type,`Screening ZEUS`,`Diagnostic ZEUS`,subtype_simple,stage)
datx&lt;-datx %&gt;% gather(key=model,value=score,-stage,-subtype_simple,-classifier_type,-id) %&gt;% filter(!is.na(score))

features&lt;-features %&gt;% filter(id %in% datx$id)

inner_join(features %&gt;% select(id,classifier_type2),datx %&gt;% select(model,id,classifier_type),by=&quot;id&quot;)</code></pre>
<pre><code>             id classifier_type2           model classifier_type
         &lt;char&gt;           &lt;char&gt;          &lt;char&gt;          &lt;char&gt;
  1: CGPLH1062P       non-cancer  Screening ZEUS         healthy
  2: CGPLH1062P       non-cancer Diagnostic ZEUS         healthy
  3: CGPLH1063P       non-cancer  Screening ZEUS         healthy
  4: CGPLH1063P       non-cancer Diagnostic ZEUS         healthy
  5: CGPLH1065P       non-cancer  Screening ZEUS         healthy
 ---                                                            
954: PGDX26277P           Benign Diagnostic ZEUS         healthy
955: PGDX26278P           Benign  Screening ZEUS         healthy
956: PGDX26278P           Benign Diagnostic ZEUS         healthy
957: PGDX26279P           Benign  Screening ZEUS         healthy
958: PGDX26279P           Benign Diagnostic ZEUS         healthy</code></pre>
<pre class="r"><code>features &lt;- features %&gt;%
        gather(key=&quot;Arm&quot;, value=&quot;Zscore&quot;, starts_with(&quot;z&quot;))
    features$Arm &lt;- sapply( str_split(features$Arm,&quot;_&quot;), &quot;[&quot;, 2 )

    chrlevels &lt;- c(&quot;1p&quot;,&quot;1q&quot;,&quot;2p&quot;,&quot;2q&quot;,&quot;3p&quot;,&quot;3q&quot;,&quot;4p&quot;,&quot;4q&quot;,&quot;5p&quot;,
                   &quot;5q&quot;,&quot;6p&quot;,&quot;6q&quot;,&quot;7p&quot;,&quot;7q&quot;,&quot;8p&quot;,&quot;8q&quot;,&quot;9p&quot;,&quot;9q&quot;,
                   &quot;10p&quot;,&quot;10q&quot;,&quot;11p&quot;,&quot;11q&quot;,&quot;12p&quot;,&quot;12q&quot;,
                   &quot;13q&quot;,&quot;14q&quot;,&quot;15q&quot;,&quot;16p&quot;,&quot;16q&quot;,&quot;17p&quot;,&quot;17q&quot;,
                   &quot;18p&quot;,&quot;18q&quot;,&quot;19p&quot;,&quot;19q&quot;,&quot;20p&quot;,&quot;20q&quot;,&quot;21q&quot;,&quot;22q&quot;)
    features$Arm &lt;- factor(features$Arm,
                           levels = chrlevels)
    features &lt;- features %&gt;% mutate(colors=if_else(Zscore&lt;=0,&quot;Negative&quot;,&quot;Positive&quot;))
    ##features$colors&lt;-factor(features$colors,levels=c(&quot;Positive&quot;,&quot;Negative&quot;))
    ##Z-score for two-tailed p-value .05 w/ bonferroni correction for 39 arms
    features &lt;- features %&gt;%
        mutate(colors=if_else(Zscore&gt;= -3.22 &amp; Zscore &lt;= 3.22 ,
                              &quot;Neutral&quot;, features$colors))
    ##features&lt;- features %&gt;% mutate(transp=if_else(colors==&quot;Neutral&quot;,.9,.2))
    ##features&lt;- features %&gt;% mutate(transp=if_else(colors==&quot;Neutral&quot;,.05,.99))
    features &lt;- features %&gt;% mutate(transp=if_else(colors==&quot;Neutral&quot;,1,1))
    ##features&lt;-features %&gt;% mutate(colors=if_else(Zscore&lt;=0,&quot;Negative&quot;,&quot;Positive&quot;))
    features$colors&lt;-factor(features$colors,levels=c(&quot;Positive&quot;,&quot;Negative&quot;,&quot;Neutral&quot;))
    features$y=1
    ##features$log&lt;-log10(abs(features$Zscore))
    ##features &lt;- features %&gt;% mutate(log=if_else(Zscore&lt;=0,-1*log,log))
    features &lt;- features %&gt;% mutate(Zscore=if_else(Zscore&lt;= -75,-75,Zscore))
    features &lt;- features %&gt;% mutate(Zscore=if_else(Zscore&gt;75,75,Zscore))
    features &lt;- features %&gt;%
        mutate(`type`=if_else(`classifier_type2`==&quot;cancer&quot;,&quot;Ovarian Cancer&quot;,classifier_type2))
      features &lt;- features %&gt;%
        mutate(`type`=if_else(`classifier_type2`==&quot;non-cancer&quot;,&quot;Non-cancer&quot;,type))

    features$type &lt;- factor(features$`type`,
                                  levels=c(&quot;Ovarian Cancer&quot;,&quot;Benign&quot;,&quot;Non-cancer&quot;))
    features2  &lt;- features %&gt;%
        rename(z=Zscore) %&gt;%
        mutate(root2=nroot(z, 2)) %&gt;%
        mutate(root3=nroot(z, 3))


features2 &lt;- features2  %&gt;%
    select(id, type, Arm,
           z, root2, root3, y, colors, transp)
###YOU ARE HERE AFTER LUNCH!!!
ns &lt;- features2 %&gt;%
    group_by(type) %&gt;%
    summarize(n=length(unique(id))) %&gt;%
    mutate(label=paste0(type, &quot;\n(n = &quot;, n, &quot;)&quot;))
ns &lt;- setNames(ns$label, ns$type)
cols[2]&lt;-&quot;#18468B&quot;
cols[1]&lt;-&quot;red3&quot;

zplot_cancer &lt;- features2 %&gt;%
    ggplot(aes(x=y, y=z, color=colors, alpha=transp)) +
    geom_point(size=5) +
    facet_grid(Arm ~ type, scales = &#39;free_y&#39;,labeller=labeller(type=ns)) +
    coord_flip() +
    theme_minimal(base_size=24) +
    theme(strip.text.y.left = element_text(angle = 0,
                                           hjust = 0.5,
                                           vjust = 0.5),
          axis.text.y = element_blank(),
          axis.ticks = element_blank(),

          panel.grid =  element_blank(),
          plot.title = element_blank(),
          legend.title = element_blank(),
          axis.title.y=element_blank(),
          legend.position = &quot;bottom&quot;,
          strip.text.y = element_blank(),
          axis.title.x = element_blank()) + #####removing label
    scale_y_continuous(limits = c(-75,75),breaks=c(-70,0,70)) +
    scale_alpha_identity() +
    scale_fill_manual(values = cols) +
    scale_color_manual(values = cols) #+
    #ylab(&quot;Z-Score&quot;)</code></pre>
<pre class="r"><code>meta&lt;-fread(here(&quot;data&quot;,&quot;metadata.csv&quot;))
meta&lt;-meta %&gt;% select(-V1) %&gt;% mutate(classifier_type2=if_else(id==&quot;CGPLOV161P&quot;,&quot;Benign&quot;,classifier_type2))

data&lt;-fread(here(&quot;data&quot;,&quot;screening_test_data.csv&quot;)) %&gt;% select(-type)
data2&lt;-fread(here(&quot;data&quot;,&quot;diag_test_data.csv&quot;)) %&gt;% filter(type==&quot;healthy&quot;) %&gt;% select(-type)
data&lt;-rbind(data,data2)

data&lt;-inner_join(data,meta %&gt;% select(id,classifier_type),by=&quot;id&quot;) %&gt;% rename(type=classifier_type)



features&lt;-data

cols &lt;- brewer.pal(7,&quot;Set1&quot;)
cols[3] &lt;- cols[4]

features &lt;- features %&gt;% select(id,starts_with(&quot;z&quot;))
features &lt;- inner_join(features, meta %&gt;%
                                     select(id,`classifier_type2`),
                           by=c(&quot;id&quot;=&quot;id&quot;))




datx&lt;-read_csv(here(&quot;data/scores&quot;,&quot;All_Scores_Val.csv&quot;))</code></pre>
<pre><code>New names:
Rows: 112 Columns: 22
â”€â”€ Column specification
â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ Delimiter: &quot;,&quot; chr
(15): id, id2, cohort, type, type2, training_set_include_screening, trai... dbl
(5): ...1, Screening ZEUS, Diagnostic ZEUS, ca125, he4 lgl (2):
menopause_status, history_of_cancer
â„¹ Use `spec()` to retrieve the full column specification for this data. â„¹
Specify the column types or set `show_col_types = FALSE` to quiet this message.
â€¢ `` -&gt; `...1`</code></pre>
<pre class="r"><code>datx&lt;-datx %&gt;% filter(screen_val==&quot;yes&quot;|diag_val==&quot;yes&quot;)
datx&lt;-datx %&gt;% select(id,classifier_type,`Screening ZEUS`,`Diagnostic ZEUS`,subtype_simple,stage)
datx&lt;-datx %&gt;% gather(key=model,value=score,-stage,-subtype_simple,-classifier_type,-id) %&gt;% filter(!is.na(score))

features&lt;-features %&gt;% filter(id %in% datx$id)

features &lt;- features %&gt;%
        gather(key=&quot;Arm&quot;, value=&quot;Zscore&quot;, starts_with(&quot;z&quot;))
    features$Arm &lt;- sapply( str_split(features$Arm,&quot;_&quot;), &quot;[&quot;, 2 )

    chrlevels &lt;- c(&quot;1p&quot;,&quot;1q&quot;,&quot;2p&quot;,&quot;2q&quot;,&quot;3p&quot;,&quot;3q&quot;,&quot;4p&quot;,&quot;4q&quot;,&quot;5p&quot;,
                   &quot;5q&quot;,&quot;6p&quot;,&quot;6q&quot;,&quot;7p&quot;,&quot;7q&quot;,&quot;8p&quot;,&quot;8q&quot;,&quot;9p&quot;,&quot;9q&quot;,
                   &quot;10p&quot;,&quot;10q&quot;,&quot;11p&quot;,&quot;11q&quot;,&quot;12p&quot;,&quot;12q&quot;,
                   &quot;13q&quot;,&quot;14q&quot;,&quot;15q&quot;,&quot;16p&quot;,&quot;16q&quot;,&quot;17p&quot;,&quot;17q&quot;,
                   &quot;18p&quot;,&quot;18q&quot;,&quot;19p&quot;,&quot;19q&quot;,&quot;20p&quot;,&quot;20q&quot;,&quot;21q&quot;,&quot;22q&quot;)
    features$Arm &lt;- factor(features$Arm,
                           levels = chrlevels)
    features &lt;- features %&gt;% mutate(colors=if_else(Zscore&lt;=0,&quot;Negative&quot;,&quot;Positive&quot;))
    ##features$colors&lt;-factor(features$colors,levels=c(&quot;Positive&quot;,&quot;Negative&quot;))
    ##Z-score for two-tailed p-value .05 w/ bonferroni correction for 39 arms
    features &lt;- features %&gt;%
        mutate(colors=if_else(Zscore&gt;= -3.22 &amp; Zscore &lt;= 3.22 ,
                              &quot;Neutral&quot;, features$colors))
    ##features&lt;- features %&gt;% mutate(transp=if_else(colors==&quot;Neutral&quot;,.9,.2))
    ##features&lt;- features %&gt;% mutate(transp=if_else(colors==&quot;Neutral&quot;,.05,.99))
    features &lt;- features %&gt;% mutate(transp=if_else(colors==&quot;Neutral&quot;,1,1))
    ##features&lt;-features %&gt;% mutate(colors=if_else(Zscore&lt;=0,&quot;Negative&quot;,&quot;Positive&quot;))
    features$colors&lt;-factor(features$colors,levels=c(&quot;Positive&quot;,&quot;Negative&quot;,&quot;Neutral&quot;))
    features$y=1
    ##features$log&lt;-log10(abs(features$Zscore))
    ##features &lt;- features %&gt;% mutate(log=if_else(Zscore&lt;=0,-1*log,log))
    features &lt;- features %&gt;% mutate(Zscore=if_else(Zscore&lt;= -75,-75,Zscore))
    features &lt;- features %&gt;% mutate(Zscore=if_else(Zscore&gt;75,75,Zscore))
    features &lt;- features %&gt;%
        mutate(`type`=if_else(`classifier_type2`==&quot;cancer&quot;,&quot;Ovarian Cancer&quot;,classifier_type2))
      features &lt;- features %&gt;%
        mutate(`type`=if_else(`classifier_type2`==&quot;non-cancer&quot;,&quot;Non-cancer&quot;,type))

    features$type &lt;- factor(features$`type`,
                                  levels=c(&quot;Ovarian Cancer&quot;,&quot;Benign&quot;,&quot;Non-cancer&quot;))
    features2  &lt;- features %&gt;%
        rename(z=Zscore) %&gt;%
        mutate(root2=nroot(z, 2)) %&gt;%
        mutate(root3=nroot(z, 3))


features2 &lt;- features2  %&gt;%
    select(id, type, Arm,
           z, root2, root3, y, colors, transp)
###YOU ARE HERE AFTER LUNCH!!!
ns &lt;- features2 %&gt;%
    group_by(type) %&gt;%
    summarize(n=length(unique(id))) %&gt;%
    mutate(label=paste0(type, &quot;\n(n = &quot;, n, &quot;)&quot;))
ns &lt;- setNames(ns$label, ns$type)
cols[2]&lt;-&quot;#18468B&quot;
cols[1]&lt;-&quot;red3&quot;

zplot_cancer_val &lt;- features2 %&gt;%
    ggplot(aes(x=y, y=z, color=colors, alpha=transp)) +
    geom_point(size=5) +
    facet_grid(Arm ~ type, scales = &#39;free_y&#39;,labeller=labeller(type=ns)) +
    coord_flip() +
    theme_minimal(base_size=24) +
    theme(strip.text.y.left = element_text(angle = 0,
                                           hjust = 0.5,
                                           vjust = 0.5),
          axis.text.y = element_blank(),
          axis.ticks = element_blank(),

          panel.grid =  element_blank(),
          plot.title = element_blank(),
          legend.title = element_blank(),
          axis.title.y=element_blank(),
          legend.position = &quot;bottom&quot;,
          strip.text.y = element_blank(),
          axis.title.x = element_blank()) + #####removing label
    scale_y_continuous(limits = c(-75,75),breaks=c(-70,0,70)) +
    scale_alpha_identity() +
    scale_fill_manual(values = cols) +
    scale_color_manual(values = cols) #+
    #ylab(&quot;Z-Score&quot;)</code></pre>
<p>Adding together the TCGA and z-score panel â€“ problems â€“ chromosome
width, size of TCGA panel, not yet stacked with feature importance
panel.</p>
<pre class="r"><code>#pdf(&quot;test.pdf&quot;)
cowplot::plot_grid(plot,zplot_cancer,zplot_cancer_val,nrow=1,align=&quot;v&quot;,axis=&quot;l&quot;)</code></pre>
<p><img src="figure/S9.Rmd/cowplot-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cowplot-1">
Past versions of cowplot-1.png
</button>
</p>
<div id="fig-cowplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/cancer-genomics/delfipro2024/blob/c74d1ee3872c077b7bd65f0ffb6aa657b10a8e94/docs/figure/S9.Rmd/cowplot-1.png" target="_blank">c74d1ee</a>
</td>
<td>
shay-279
</td>
<td>
2024-08-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off() #</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin23.4.0
Running under: macOS Sonoma 14.6

Matrix products: default
BLAS:   /opt/homebrew/Cellar/openblas/0.3.27/lib/libopenblasp-r0.3.27.dylib 
LAPACK: /opt/homebrew/Cellar/r/4.4.1/lib/R/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] liver.tools_0.0.0.9000 devtools_2.4.5         usethis_3.0.0         
 [4] readxl_1.4.3           ggstance_0.3.7         here_1.0.1            
 [7] gridExtra_2.3          RColorBrewer_1.1-3     data.table_1.15.4     
[10] lubridate_1.9.3        forcats_1.0.0          stringr_1.5.1         
[13] dplyr_1.1.4            purrr_1.0.2            readr_2.1.5           
[16] tidyr_1.3.1            tibble_3.2.1           ggplot2_3.5.1         
[19] tidyverse_2.0.0        workflowr_1.7.1       

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1    farver_2.1.2        fastmap_1.2.0      
 [4] promises_1.3.0      digest_0.6.36       timechange_0.3.0   
 [7] mime_0.12           lifecycle_1.0.4     ellipsis_0.3.2     
[10] processx_3.8.4      magrittr_2.0.3      compiler_4.4.1     
[13] rlang_1.1.4         sass_0.4.9          tools_4.4.1        
[16] utf8_1.2.4          yaml_2.3.10         knitr_1.48         
[19] labeling_0.4.3      htmlwidgets_1.6.4   bit_4.0.5          
[22] pkgbuild_1.4.4      pkgload_1.4.0       miniUI_0.1.1.1     
[25] withr_3.0.1         desc_1.4.3          fansi_1.0.6        
[28] urlchecker_1.0.1    git2r_0.33.0.9000   profvis_0.3.8      
[31] xtable_1.8-4        colorspace_2.1-1    scales_1.3.0       
[34] cli_3.6.3           rmarkdown_2.27      crayon_1.5.3       
[37] generics_0.1.3      remotes_2.5.0       rstudioapi_0.16.0  
[40] httr_1.4.7          tzdb_0.4.0          sessioninfo_1.2.2  
[43] cachem_1.1.0        parallel_4.4.1      BiocManager_1.30.23
[46] cellranger_1.1.0    vctrs_0.6.5         jsonlite_1.8.8     
[49] callr_3.7.6         hms_1.1.3           bit64_4.0.5        
[52] jquerylib_0.1.4     glue_1.7.0          ps_1.7.7           
[55] cowplot_1.1.3       stringi_1.8.4       gtable_0.3.5       
[58] later_1.3.2         munsell_0.5.1       pillar_1.9.0       
[61] htmltools_0.5.8.1   R6_2.5.1            rprojroot_2.0.4    
[64] vroom_1.6.5         evaluate_0.24.0     shiny_1.9.1        
[67] highr_0.11          memoise_2.0.1       renv_1.0.7         
[70] httpuv_1.6.15       bslib_0.8.0         Rcpp_1.0.13        
[73] whisker_0.4.1       xfun_0.46           fs_1.6.4           
[76] getPass_0.2-4       pkgconfig_2.0.3    </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
